android-components/components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonFilePicker.kt

```diff
@@ -52,13 +56,15 @@ class AddonFilePicker(
     }
 
     internal fun handleUriSelected(uri: Uri?) {
-        uri?.let {
+        uri?.let { safeUri ->
             val fileUri = convertToFileUri(uri)
```

Comment:
Can you help me understand why we have:

- `uri`
- `safeUri`
- `fileUri`

in these two lines?

Reply:
Sure, 

`uri` -> It's the null able uri we get from Andriod after selecting the file.
`safeUri` -> it's non null version, which comes from `let`.
`fileUri`-> it's a string file `uri` version that comes from converting `uri` a file version.

Reply:
Why don't we pass `safeUri` to `convertToFileUri()`?

Reply:
We are implicitly doing it :) , as we are inside of the `let?.` block we are warranty that `uri` will be no-null so the compiler does an implicit cast. I update the code to make it explicit :)  
<img width="739" alt="image" src="https://github.com/mozilla-mobile/firefox-android/assets/773158/6e55992b-aa3a-4a51-822f-4953d66aa7ca">